modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -349,19 +353,102 @@ def get_dataset(cls, path, engine, storage_options):
             raise ValueError("engine must be one of 'pyarrow', 'fastparquet'")
 
     @classmethod
-    def call_deploy(cls, dataset, col_partitions, storage_options, **kwargs):
+    def _determine_partitioning(
+        cls, dataset: ColumnStoreDataset
+    ) -> "list[list[ParquetFileToRead]]":
+        """
+        Determine which partition will read certain files/row groups of the dataset.
+
+        Parameters
+        ----------
+        dataset : ColumnStoreDataset
+
+        Returns
+        -------
+        list[list[ParquetFileToRead]]
+            Each element in the returned list describes a list of files that a partition has to read.
+        """
+        from modin.core.storage_formats.pandas.parsers import ParquetFileToRead
```

Comment:
why cannot we import it at the top without `if TYPE_CHECKING` condition?

Reply:
it then triggers "an import from a partially initialized module" problem